.github/workflows/main.yml

```diff
@@ -48,4 +48,24 @@ jobs:
       - name: Run basic yaml validation
         run: |
           python tests/validate_yaml.py
-        
+  on-fail:
+    if: failure()
+    runs-on: ubuntu-latest
+    needs: check
+    steps:
+      - name: Add label to PR
+        uses: actions/github-script@v7
+        with:
+          script: |
+            const label = 'staging-skip';
+            const { context, github } = require('@actions/github');
+            const owner = context.repo.owner;
+            const repo = context.repo.repo;
+            const pull_number = context.payload.pull_request.number;
+
+            await github.issues.addLabels({
```

Comment:
Adding labels requires changing the repo settings: Actions -> General -> Workflow Permissions -> Read and Write permissions